error[E0277]: the type `std::collections::HashSet<{integer}>` cannot be indexed by `usize`
  --> $DIR/macro_fail_bad_type.rs:12:5
   |
12 |     comp!([for i in 1..20 => i], HashSet);         //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     `std::collections::HashSet<{integer}>` cannot be indexed by `usize`
   |     required by a bound in this
   |     required by this bound in `test_vec_wrong_types::_seq`
   |
   = help: the trait `std::ops::Index<usize>` is not implemented for `std::collections::HashSet<{integer}>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the type `std::collections::BTreeSet<{integer}>` cannot be indexed by `usize`
  --> $DIR/macro_fail_bad_type.rs:13:5
   |
13 |     comp!([for i in 1..20 => i], BTreeSet);        //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     `std::collections::BTreeSet<{integer}>` cannot be indexed by `usize`
   |     required by a bound in this
   |     required by this bound in `test_vec_wrong_types::_seq`
   |
   = help: the trait `std::ops::Index<usize>` is not implemented for `std::collections::BTreeSet<{integer}>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0107]: wrong number of type arguments: expected at least 2, found 1
  --> $DIR/macro_fail_bad_type.rs:14:5
   |
14 |     comp!([for i in 1..20 => i], HashMap);         //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected at least 2 type arguments
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0107]: wrong number of type arguments: expected 2, found 1
  --> $DIR/macro_fail_bad_type.rs:15:5
   |
15 |     comp!([for i in 1..20 => i], BTreeMap);        //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 2 type arguments
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: no implementation for `&std::vec::Vec<{integer}> & &std::vec::Vec<{integer}>`
  --> $DIR/macro_fail_bad_type.rs:20:5
   |
20 |     comp!({for i in 1..20 => i}, Vec);            //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     no implementation for `&std::vec::Vec<{integer}> & &std::vec::Vec<{integer}>`
   |     required by a bound in this
   |     required by this bound in `test_set_wrong_types::_st`
   |
   = help: the trait `std::ops::BitAnd` is not implemented for `&std::vec::Vec<{integer}>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: no implementation for `&std::vec::Vec<{integer}> | &std::vec::Vec<{integer}>`
  --> $DIR/macro_fail_bad_type.rs:20:5
   |
20 |     comp!({for i in 1..20 => i}, Vec);            //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     no implementation for `&std::vec::Vec<{integer}> | &std::vec::Vec<{integer}>`
   |     required by a bound in this
   |     required by this bound in `test_set_wrong_types::_st`
   |
   = help: the trait `std::ops::BitOr` is not implemented for `&std::vec::Vec<{integer}>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: no implementation for `&std::vec::Vec<{integer}> ^ &std::vec::Vec<{integer}>`
  --> $DIR/macro_fail_bad_type.rs:20:5
   |
20 |     comp!({for i in 1..20 => i}, Vec);            //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     no implementation for `&std::vec::Vec<{integer}> ^ &std::vec::Vec<{integer}>`
   |     required by a bound in this
   |     required by this bound in `test_set_wrong_types::_st`
   |
   = help: the trait `std::ops::BitXor` is not implemented for `&std::vec::Vec<{integer}>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: no implementation for `&std::collections::VecDeque<{integer}> & &std::collections::VecDeque<{integer}>`
  --> $DIR/macro_fail_bad_type.rs:21:5
   |
21 |     comp!({for i in 1..20 => i}, VecDeque);       //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     no implementation for `&std::collections::VecDeque<{integer}> & &std::collections::VecDeque<{integer}>`
   |     required by a bound in this
   |     required by this bound in `test_set_wrong_types::_st`
   |
   = help: the trait `std::ops::BitAnd` is not implemented for `&std::collections::VecDeque<{integer}>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: no implementation for `&std::collections::VecDeque<{integer}> | &std::collections::VecDeque<{integer}>`
  --> $DIR/macro_fail_bad_type.rs:21:5
   |
21 |     comp!({for i in 1..20 => i}, VecDeque);       //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     no implementation for `&std::collections::VecDeque<{integer}> | &std::collections::VecDeque<{integer}>`
   |     required by a bound in this
   |     required by this bound in `test_set_wrong_types::_st`
   |
   = help: the trait `std::ops::BitOr` is not implemented for `&std::collections::VecDeque<{integer}>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: no implementation for `&std::collections::VecDeque<{integer}> ^ &std::collections::VecDeque<{integer}>`
  --> $DIR/macro_fail_bad_type.rs:21:5
   |
21 |     comp!({for i in 1..20 => i}, VecDeque);       //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |
   |     no implementation for `&std::collections::VecDeque<{integer}> ^ &std::collections::VecDeque<{integer}>`
   |     required by a bound in this
   |     required by this bound in `test_set_wrong_types::_st`
   |
   = help: the trait `std::ops::BitXor` is not implemented for `&std::collections::VecDeque<{integer}>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0107]: wrong number of type arguments: expected at least 2, found 1
  --> $DIR/macro_fail_bad_type.rs:22:5
   |
22 |     comp!({for i in 1..20 => i}, HashMap);        //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected at least 2 type arguments
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0107]: wrong number of type arguments: expected 1, found 2
  --> $DIR/macro_fail_bad_type.rs:31:5
   |
31 |     comp!({for i in 1..20 => i, i}, Vec);         //~ ERROR
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unexpected type argument
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 12 previous errors

Some errors have detailed explanations: E0107, E0277.
For more information about an error, try `rustc --explain E0107`.
